compact.go

```diff
@@ -476,9 +486,14 @@ func (w *instrumentedChunkWriter) WriteChunks(chunks ...chunks.Meta) error {
 func (c *LeveledCompactor) write(dest string, meta *BlockMeta, blocks ...BlockReader) (err error) {
 	dir := filepath.Join(dest, meta.ULID.String())
 	tmp := dir + ".tmp"
-
+	var writers []io.Closer
+	var merr MultiError
 	defer func(t time.Time) {
-		if err != nil {
+		merr.Add(err)
```

Comment:
We need different MultiError here (: as your sometimes return `merr.Err` which would we added here then.